packages/preview/src/browser/preview-handler.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, named } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { ContributionProvider } from "@theia/core";
+
+export const PreviewHandler = Symbol('PreviewHandler');
+
+export interface PreviewHandler {
+    canHandle(uri: URI): boolean;
+    renderHTML(content: string): Promise<string | undefined>;
```

Comment:
please use `MaybePromise` in APIs that extenders don't need to wrap results in promises if not needed